Jenkinsfile

```diff
@@ -20,137 +20,172 @@ pipeline {
+            stage("make cyclomatic-check") {
+              steps {
+                dir(path: "$REPO_DIR") {
+                  sh "make cyclomatic-check"
+                }
+              }
+            }
+            stage("make test BUILDTAGS=kerneldrv") {
+              steps {
+                dir(path: "$REPO_DIR") {
+                  sh "make test BUILDTAGS=kerneldrv"
+                }
+              }
             }
           }
         }
-        stage("make images with buildah") {
+        stage('make pre-pull') {
           steps {
-            dir(path: "$REPO_DIR") {
-              sh "make images BUILDER=buildah"
+    	      dir(path: "$REPO_DIR") {
+              sh "make pre-pull"
             }
           }
         }
-      }
-    }
-    stage('make demos') {
-      parallel {
-        stage('make demos with docker') {
-          steps {
-            dir(path: "$REPO_DIR") {
-              sh "make demos"
+        stage('make images') {
+          parallel {
+            stage("make images with docker") {
+              steps {
+                dir(path: "$REPO_DIR") {
+                  sh "make -j4 images"
+                }
+              }
+            }
+            stage("make images with buildah") {
+              steps {
+                dir(path: "$REPO_DIR") {
+                  sh "make images BUILDER=buildah"
+                }
+              }
             }
           }
         }
-        stage('make demos with buildah') {
-          steps {
-            dir(path: "$REPO_DIR") {
-              sh "make demos BUILDER=buildah"
+        stage('make demos') {
+          parallel {
+            stage('make demos with docker') {
+              steps {
+                dir(path: "$REPO_DIR") {
+                  sh "make demos"
+                }
+              }
+            }
+            stage('make demos with buildah') {
+              steps {
+                dir(path: "$REPO_DIR") {
+                  sh "make demos BUILDER=buildah"
+                }
+              }
             }
           }
         }
       }
-    }
-  }
-  post {
-    success {
-      script {
-        if (env.CHANGE_ID == null) {
+      post {
+        success {
           withDockerRegistry([ credentialsId: "e16bd38a-76cb-4900-a5cb-7f6aa3aeb22d", url: "https://${REG}" ]) {
             sh "make push"
           }
         }
       }
     }
+    stage('Intel Device plugins') {
+      when { changeRequest() }
+      agent {
+        label "clr-bmaas-intel-device-plugins"
+      }
+      environment {
+        WORKDIR="${env.WORKSPACE}/scripts/jenkins"
+      }
+      stages {
+        stage('Tests') {
+          parallel {
+            stage('QAT') {
+              steps {
+                dir(path: "$WORKDIR/qat") {
+                  withCredentials([usernamePassword(credentialsId: 'e16bd38a-76cb-4900-a5cb-7f6aa3aeb22d', passwordVariable: 'RPASS', usernameVariable: 'RUSER')]) {
+                    sh 'make qat'
```

Comment:
One comment I made earlier was that could we just have `make tests` and then `Makefile` builds the test targets to run based on, e.g., plugin specific implementations found in `test/*` directory.